[BUG] - Update timeseries range checks

TomDonoghue opened the pull request:
Responds to #217 

The current update updates the `SingleTracePlotyWidget` to deal with displaying a selected segment that happens to be empty. 

Updates:
- add a try / except for setting y-axis limits, and default to None if no data
- update x-axis limits to show the selected range, rather than the x-range from the data
    - Note: this changes behaviour a bit: previously updating data would autoscale to the range of the selected data (so if selecting 30 seconds, but data was only in range 5-25, it would show 5-25), whereas with this update the x-range will always match the selected range. This is an easy way to address this specific issue, but might not be a preferred way to do it? (An alternative is to add an extra try / except, and show empty range if no data, auto-scale if there is _any_ data (?))

Notes:
- if this solution is considered a good way to go, I think it probably wants to be similarly updated in `SeparateTracesPlotlyWidget` (`on_change` - lines ~398-421).

bendichter commented:
This looks like it would probably work, but I prefer to avoid try/except flows. Would it be possible to implement this by testing of yy is empty instead?

TomDonoghue commented:
Comments / updates integrated!

Quick notes:
- there was a duplicated section in the timeseries tests, so one commit cleans that up
- at the end I added the same procedure to the multi-timeseries widget, so that the two widgets act the same
- tests are added: i added timestamps to have a gap and then selected an empty time range
